gemonlinedb/src/common/FileConfigurationProvider.cc

```diff
@@ -136,80 +92,116 @@ namespace gem {
             }
         }
 
-        void XMLConfigurationProvider::loadAMC13(const std::string &filename)
+        void FileConfigurationProvider::loadAMC13(const std::string &filename)
         {
             // Find the file (throws if not found)
             auto path = detail::getFileInPath(filename, getSearchPath());
             m_sources.push_back(path);
 
-            detail::XercesGuard guard; // Make sure that Xerces is loaded
-            auto document = loadDOM(path, "AMC13Configuration.xsd");
-
-            loadInternal(path, document, m_amc13Config);
+            if (boost::algorithm::ends_with(path, ".json")) {
```

Comment:
I figured that checking the extension could be a useful semantic check (to check that the user isn't trying to load XML data or a GBT blob file...).
When `loadInternal` fails it throws `xcept::Exception` or `std::exception` derived classes.

Reply:
right, but i could easily just have my blob/xml named `myxmlfile.json` (for unknown/nefarious reasons) and the check will pass, though it *should* still fail in the `loadInternal` call.
likewise, my valid `JSON` could be named `myjson.txt`, but this would fail.
so i was just wondering:
1) if the `boost` dependency was truly necessary (this was where i started, even though i know there are many other `boost` related things scattered throughout the code)
1) if the code could be simplified and rely on the `loadInternal` to handle invalid input data

Reply:
Code simplified and `boost` includes dropped.